pkg/registry/generic/registry/store.go

```diff
@@ -501,10 +502,15 @@ var (
 // shouldUpdateFinalizers returns if we need to update the finalizers of the
 // object, and the desired list of finalizers.
 // When deciding whether to add the OrphanDependent finalizer, factors in the
-// order of highest to lowest priority are: options.OrphanDependents, existing
-// finalizers of the object, e.DeleteStrategy.DefaultGarbageCollectionPolicy.
+// order of highest to lowest priority are: options.OrphanDependents,
+// options.SynchronousGC, existing finalizers of the object,
+// e.DeleteStrategy.DefaultGarbageCollectionPolicy.
+// When deciding whether to add the SynchronousGC finalizer, factors in the
+// order of highest to lowest priority are: options.SynchronousGC, existing
+// finalizers of the object, default to false.
 func shouldUpdateFinalizers(e *Store, accessor meta.Object, options *api.DeleteOptions) (shouldUpdate bool, newFinalizers []string) {
```

Comment:
finalizers is likely to be a very short list.  Any way to refactor this for a multi-pass with priorities to keep the orphan and the synchronous gc logic separate?